src/neuro_flow/expr.py

```diff
@@ -1367,6 +1367,53 @@ class URIExpr(URIExprMixin, StrictExpr[URL]):
     pass
 
 
+class PlatformResourceURIExpr(URIExpr):
+    async def eval(self, root: RootABC) -> URL:
+        ret = await super().eval(root)
+        try:
+            # Hack to get flow's project ctx -> project_name
+            project_ctx = root.lookup("project")
+        except LookupError:
+            project_ctx = None
+
+        if project_ctx:
+            async with root.client() as cl:
+                ret = cl.parse.str_to_uri(
+                    str(ret),
+                    project_name=project_ctx.project_name,  # type: ignore
+                    short=True,
+                )
+        return ret
+
+
+class OptImageRefStrExpr(OptStrExpr):
+    async def eval(self, root: RootABC) -> Optional[str]:
+        ret = await super().eval(root)
+        project_ctx = None
+        if ret and ret.startswith("image:"):
+            try:
+                # Hack to get flow's project ctx -> project_name
+                project_ctx = root.lookup("project")
+            except LookupError:
+                pass
+        if ret and project_ctx:
+            async with root.client() as cl:
+                uri = cl.parse.str_to_uri(
+                    ret,
+                    project_name=project_ctx.project_name,  # type: ignore
+                    short=True,
+                )
+                ret = str(uri)
+        return ret
+
+
+class ImageRefStrExpr(OptImageRefStrExpr):
+    async def eval(self, root: RootABC) -> str:
+        ret = await super().eval(root)
+        assert ret
```

Comment:
if the idea is to fail here, we should rely on an explicit exception as `assert`s may be disabled in run time.

Reply:
it's actually to convert Optional Image ref to image ref, but that's was more like a hack, done in a proper way in [f34f806](https://github.com/neuro-inc/neuro-flow/pull/1081/commits/f34f806f3a0b29dbe0be3ba5d0b76dfce494b381)